distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestSingleSignOn.java

```diff
@@ -87,182 +113,249 @@ public void beforeTest() throws Exception {
+        // Start the services needed for testing.
+        // We need to start the Search UI to test that it redirects properly
         getServiceManager().startFeature(true, "security-idp", "search-ui-app");
 
-        // Create a validator from the metadata schema so that we can validate the metadata
-        SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
-        Schema schema = schemaFactory.newSchema(getClass().getClassLoader().getResource("saml-schema-metadata-2.0.xsd"));
-        Validator validator = schema.newValidator();
-
-        // Get the metadata
-        String serverMetadata = get(SERVICE_ROOT + "/idp/login/metadata").asString();
-        String clientMetadata = get(SERVICE_ROOT + "/saml/sso/metadata").asString();
-
-        // Ensure the metadata is valid according to the SAML Metadata schema
-        validator.validate(new StreamSource(new StringReader(clientMetadata)));
-        validator.validate(new StreamSource(new StringReader(serverMetadata)));
-
-        // To find the right inputs for the settings, go into the metatype.xml file.
-        // The key is the "id" and the value type is determined by the cardinality as such:
-        // Positive = array, negative = vector, 0 (none) = single variable
-        Dictionary<String, Object> clientSettings = new Hashtable<>();
-        clientSettings.put("metadata", serverMetadata);
-        Dictionary<String, Object> serverSettings = new Hashtable<>();
-        serverSettings.put("spMetadata", new String[] {clientMetadata});
-
-        // Update the client and server with the metadata
-        Configuration clientConfig = getAdminConfig()
-                .getConfiguration("org.codice.ddf.security.idp.client.IdpMetadata", null);
-        Configuration serverConfig = getAdminConfig()
-                .getConfiguration("org.codice.ddf.security.idp.server.IdpEndpoint", null);
-        clientConfig.update(clientSettings);
-        serverConfig.update(serverSettings);
-        expect("Configs to update").within(2, TimeUnit.MINUTES)
-                .until(clientConfig::getProperties, notNullValue());
-        expect("Configs to update").within(2, TimeUnit.MINUTES)
-                .until(serverConfig::getProperties, notNullValue());
+        // Get all of the metadata
+        String metadata = get(SERVICE_ROOT + "/idp/login/metadata").asString();
+        String ddfSpMetadata = get(SERVICE_ROOT + "/saml/sso/metadata").asString();
+
+        // Make sure all the metadata is valid before we set it
+        validateSaml(metadata, SamlSchema.METADATA);
+        validateSaml(ddfSpMetadata, SamlSchema.METADATA);
+
+        // Set the metadata
+        setConfig("org.codice.ddf.security.idp.client.IdpMetadata", "metadata", metadata);
+        setConfig("org.codice.ddf.security.idp.server.IdpEndpoint", "spMetadata", new String[] {ddfSpMetadata});
```

Comment:
Ah! Yes, this would be great information to know... I'm not sure just comments would be good here, they might get ugly if lengthy... I might suggest (hopefully others don't kill me for heresy) adding javadoc to the setConfig method here.